File: ShaderUnit/TestScripts/CustomMips.py
# Test custom mipmap generation.
import utils

utils.TestSetting('sampler', 'linear', globals())
utils.TestSetting('mip', 1, globals())
filename = 'CustomDownsample_2D_' + sampler + '.hlsl'
tex = ri.LoadTexture('Assets/Textures/ThisIsATest.png', generateMips = filename)
utils.TestTextureLevel(ri, tex, mip)

File: SRPTests/TestScripts/CS/CustomMips.cs
// Test custom mipmap generation.
#load "Utils/utils.cs"

var sampler = TestSetting<string>("sampler", "linear");
var mip = TestSetting<long>("mip", 1);
var filename = "CustomDownsample_2D_" + sampler + ".hlsl";
var tex = ri.LoadTexture("Assets/Textures/ThisIsATest.png")
	.WithCustomMips(filename);

TestTextureLevel(tex, mip);
